src/content/en/fundamentals/primers/service-workers/the-players.markdown

```diff
@@ -0,0 +1,41 @@
+---
+layout: shared/narrow
+published: false
+title: "The Players"
+description: "A basic service worker implementation has two types of players."
+published_on: 2015-10-01
+updated_on: 2015-10-01
+order: 3
+translation_priority: 0
+authors:
+- josephmedley
+---
+
+<p class="intro">
+  A basic service worker implementation has two types of players.
+</p>
+
+ 
+### Clients
+
+These players install and use a service worker. Generally, they  are web
+pages. Though there are [other types of clients](https://developer.mozilla.org/en-US/docs/Web/API/Client)
+,  this primer focuses on web page clients.
+
+### Service worker script
+
+This player is a background script that acts as a network proxy for one or
+more clients. Though its name implies activity, you can almost think of a
+service worker as a passive player, sitting around waiting for its events to
+be called. It can't even manipulate  its clients' DOMs.
```

Comment:
If you're going to point out that you can't do DOM manipulation, you could counterbalance that with the things that a service worker can do:

- Access cached `Response` objects via the Cache Storage API. (We're pretty far into the primer without having mentioned the Cache Storage API once, despite it being quite important.)
- Access IndexedDB.
- Two-way communication with the clients it controls via `postMessage()`.